tests/foreman/cli/test_host.py

```diff
@@ -70,1769 +65,1497 @@
-        host = make_fake_host({'domain-id': domain['id']})
-        HostInterface.create(
-            {'host-id': host['id'], 'domain-id': domain['id'], 'mac': mac, 'type': 'interface'}
-        )
-        host = Host.info({'id': host['id']})
-        host_interface = HostInterface.info(
-            {
-                'host-id': host['id'],
-                'id': [ni for ni in host['network-interfaces'] if ni['mac-address'] == mac][0][
-                    'id'
-                ],
-            }
-        )
-        self.assertEqual(host_interface['domain'], domain['name'])
-        self.assertEqual(host_interface['mac-address'], mac)
-
-    @pytest.mark.run_in_one_thread
-    @pytest.mark.tier2
-    def test_positive_create_and_update_with_content_source(self):
-        """Create a host with content source specified and update content
-            source
-
-        :id: 5712f4db-3610-447d-b1da-0fe461577d59
-
-        :customerscenario: true
-
-        :BZ: 1260697, 1483252, 1313056, 1488465
-
-        :expectedresults: A host is created with expected content source
-            assigned and then content source is successfully updated
-
-        :CaseImportance: High
-        """
-        content_source = Proxy.list({'search': f'url = https://{settings.server.hostname}:9090'})[
-            0
-        ]
-        host = make_fake_host(
-            {
-                'content-source-id': content_source['id'],
-                'content-view-id': self.DEFAULT_CV['id'],
-                'lifecycle-environment-id': self.LIBRARY['id'],
-                'organization': self.new_org['name'],
-            }
-        )
-        self.assertEqual(
-            host['content-information']['content-source']['name'], content_source['name']
-        )
-        new_content_source = make_proxy()
-        self.addCleanup(capsule_cleanup, new_content_source['id'])
-        self.addCleanup(Host.delete, {'id': host['id']})
-        Host.update({'id': host['id'], 'content-source-id': new_content_source['id']})
-        host = Host.info({'id': host['id']})
-        self.assertEqual(
-            host['content-information']['content-source']['name'], new_content_source['name']
-        )
-
-    @pytest.mark.tier2
-    def test_negative_create_with_content_source(self):
-        """Attempt to create a host with invalid content source specified
-
-        :id: d92d6aff-4ad3-467c-88a8-5a5e56614f58
+@pytest.fixture(scope="module")
+def module_default_proxy():
+    """ Use the default installation smart proxy """
+    return Proxy.list({'search': f'url = https://{settings.server.hostname}:9090'})[0]
+
+
+@pytest.fixture(scope="module")
+def module_promoted_cv(module_lce, module_published_cv):
+    promote(module_published_cv.version[0], environment_id=module_lce.id)
+    return module_published_cv
+
+
+@pytest.fixture(scope="function")
+def function_proxy():
+    proxy = make_proxy()
+    yield proxy
+    Proxy.delete({'id': proxy['id']})
```

Comment:
These could go in the shared api fixtures.

Reply:
`module_default_proxy` and `function_proxy` are using Proxy which is CLI object
I will move `module_promoted_cv` :+1:

Reply:
@tstrych Be wary of #8197 because @swadeley also implemented `module_promoted_cv` in his PR so there might be a conflict.  Please come to an agreement to use one form of `module_promoted_cv`.

Reply:
Thanks :)